src/AndersonAcceleration.f90

```diff
@@ -72,183 +71,288 @@ MODULE AndersonAccelerationTypes
     PROCEDURE,PASS :: reset => reset_AndersonAccelerationType
 ENDTYPE AndersonAccelerationType
 
-!> Exception Handler for use in Anderson Acceleration
-TYPE(ExceptionHandlerType),SAVE :: eAndersonAccelerationType
-
 !> Name of module
-CHARACTER(LEN=*),PARAMETER :: modName='ANDERSONACCELERATIONTYPE'
-
+CHARACTER(LEN=*),PARAMETER :: modName='AndersonAccelerationTypes'
 !
 !===============================================================================
 CONTAINS
 !
 !-------------------------------------------------------------------------------
-!> @brief Initializes the Anderson Acceleration Type with a parameter list
+!> @brief Initializes the Anderson Acceleration Type with an input parameter list
 !> @param pList the parameter list
+!> @param solver The Anderson Acceleration solver to act on
+!> @param Params The Anderson options parameter list
+!> @param ce The computing environment to use for the calculation
 !>
-!> @param solver The anderson acceleration solver to act on
-!> @param MPIEnv The MPI environment description
-!> @param Params The aprameterlist
-!>
-!> This routine initializes the data spaces for the NOX Anderson Acceleration
-!>
-SUBROUTINE init_AndersonAccelerationType(solver,MPIEnv,Params)
+SUBROUTINE init_AndersonAccelerationType(solver,ce,Params)
   CHARACTER(LEN=*),PARAMETER :: myName='init_AndersonAccelerationType'
   CLASS(AndersonAccelerationType),INTENT(INOUT) :: solver
-  TYPE(MPI_EnvType),INTENT(IN),TARGET :: MPIEnv
+  TYPE(FutilityComputingEnvironment),TARGET,INTENT(IN) :: ce
   TYPE(ParamType),INTENT(IN) :: Params
 
-  TYPE(ParamType) :: validParams
-  INTEGER(SIK) :: n,nlocal,depth,start
-  REAL(SRK) :: beta
-#ifdef FUTILITY_HAVE_Trilinos_NOX
-  TYPE(ParamType) :: tmpPL
-#endif
-
-  !Check to set up required and optional param lists.
-  !IF(.NOT.AndersonType_Paramsflag) CALL AndersionType_Declare_ValidParams()
-  !Validate against the reqParams and OptParams
-  validParams=Params
-  !CALL validParams%validate(AndersonType_reqParams)
+  TYPE(ParamType) :: LSparams
+  INTEGER(SIK) :: i,j,m
 
-  IF(.NOT. MPIEnv%isInit()) THEN
-    CALL eAndersonAccelerationType%raiseError('Incorrect input to '// &
-        modName//'::'//myName//' - MPI Environment is not initialized!')
-  ELSE
-    solver%MPIparallelEnv => MPIEnv
-  ENDIF
+  REQUIRE(Params%has('AndersonAccelerationType->n'))
+  REQUIRE(Params%has('AndersonAccelerationType->depth'))
+  REQUIRE(Params%has('AndersonAccelerationType->beta'))
+  REQUIRE(Params%has('AndersonAccelerationType->start'))
```

Comment:
Just out of curiosity, is it possible to set reasonable defaults for some of these?  Or are the values fairly application-specific?

Reply:
So, defaults are/will be set in XMLProc for a normal run by a user. Depth=1, beta=0.5, start=1 is a conservative choice that should at least work "okay" for essentially any problem.

Reply:
Ok.  I'm kind of in favor of setting these defaults internally as opposed to using XMLProc... thoughts @stimpsonsg @bscollin ?

Reply:
Also, is this something that would even be set up in XMLProc?  I feel like at most someone would simply enable anderson acceleration for a particular application, and then the code would have the appropriate values for that application.

Reply:
Any change in the problem (different geometry, power, flow, anything) will lead to a change in what's optimal for depth and beta (especially beta). The default for beta should be 0.5 because that is safe (beta acts as a damping factor), but it is likely far from optimal for most problems. So, really the user should be aware of what Anderson is set to if they are going to use it.

Reply:
ok, that's fair.  But, for any of these values that could have a default, let's set a safe one here and then we can expose those to the user as needed (instead of ALWAYS requiring them to be set).  `beta` sounds like it could just default to 0.5.  It seems like `start` and `depth` could have some reasonable defaults too?  Obviously `n` has to be passed in, so that one is fine to REQUIRE the presence of.

Reply:
I'm good with that. I'll remove the defaults from XML_Proc on the MPACT side, but still warn for missing parameters there so the warnings are printed in the input processing. Does that sound good? Start can always be 1 (only time higher should really be chosen is when you know your initial guess is either garbage or not actually used for the vector you're accelerating such as when the initial vector is just defaulted to something). For depth 1 is safest, higher values for depth result in more aggressive acceleration that may or may not be stable or even faster.

Reply:
Yeah that sounds good.  Thanks